photon-server/src/main/java/org/photonvision/Main.java

```diff
@@ -347,6 +347,27 @@ private static void addTestModeSources() {
     }
 
     public static void main(String[] args) {
+        logger.info(
+                "Starting PhotonVision version "
+                        + PhotonVersion.versionString
+                        + " on "
+                        + Platform.getPlatformName()
+                        + (Platform.isRaspberryPi() ? (" (Pi " + PiVersion.getPiVersion() + ")") : ""));
+
+        try {
+            if (!handleArgs(args)) {
+                System.exit(0);
+            }
+        } catch (ParseException e) {
+            logger.error("Failed to parse command-line options!", e);
+        }
+
+        // We don't want to trigger an exit in test mode. This is specifically for MacOS.
+        if (!Platform.isSupported() && !isSmoketest) {
```

Comment:
Smoke test and test mode are two different things. We want this check to be skipped on both.